src/moneypenny/moneypenny.py

```diff
@@ -114,22 +118,30 @@ async def dispatch_order(self, action: str, dossier: Dossier) -> bool:
         container_started : `bool`
             Whether a container needed to be started for this order.  If this
             returns `False`, the action should be considered already
-            complete.
+            complete.  This might be because the order is empty, or it might
+            be because the order has already happened for this dossier and
+            action.
         """
         username = dossier.username
+        cache_key = f"{action}:{dossier}"
```

Comment:
Ah, relying on Pydantic to provide a consistent `__str__`. Yup, that looks like it should work.